CS 410 Assignment 3/src/ass4/ControllerNotePad.java

```diff
@@ -0,0 +1,222 @@
+    public void actionPerformed(ActionEvent e) {
+    	    	
+        if(e.getActionCommand().equals("new")) {
+        	model.getTextArea().setText("");
+        }else if(e.getActionCommand().equals("save")) {
+            File fileToWrite = null;
+            JFileChooser fc = new JFileChooser();
+            int returnVal = fc.showSaveDialog(null);
+            if (returnVal == JFileChooser.APPROVE_OPTION)
+                fileToWrite = fc.getSelectedFile();
+            try {
+                PrintWriter out = new PrintWriter(new FileWriter(fileToWrite));
+                out.println(model.getTextArea().getText());
+                JOptionPane.showMessageDialog(null, "File is saved successfully...");
+                out.close();
+            } catch (IOException ex) {
+            }
+        }else if(e.getActionCommand().equals("print")) {
+            try{
+                PrinterJob pjob = PrinterJob.getPrinterJob();
+                pjob.setJobName("Sample Command Pattern");
+                pjob.setCopies(1);
+                pjob.setPrintable(new Printable() {
+                    public int print(Graphics pg, PageFormat pf, int pageNum) {
+                        if (pageNum>0)
+                            return Printable.NO_SUCH_PAGE;
+                        pg.drawString(model.getTextArea().getText(), 500, 500);
+                        paint(pg);
+                        return Printable.PAGE_EXISTS;
+                    }
+                }
+                
+                		);
+
+                if (pjob.printDialog() == false)
+                    return;
+                pjob.print();
+            } catch (PrinterException pe) {
+                JOptionPane.showMessageDialog(null,
+                        "Printer error" + pe, "Printing error",
+                        JOptionPane.ERROR_MESSAGE);
+            }
+        }else if(e.getActionCommand().equals("copy")) {
+        	model.getTextArea().copy();
+        }else if(e.getActionCommand().equals("paste")) {
+            StyledDocument doc = model.getTextArea().getStyledDocument();
+            Position position = doc.getEndPosition();
+            System.out.println("offset"+position.getOffset());
+            model.getTextArea().paste();
+            
+        }
+        
+        else if(e.getActionCommand().equals("open")) {
+        	File fileToOpen = null;
+            JFileChooser fc = new JFileChooser();
+            int returnVal = fc.showOpenDialog(null);
+            if (returnVal == JFileChooser.APPROVE_OPTION)
+                fileToOpen = fc.getSelectedFile();
+            try {
+            	    
+            	Desktop desk=Desktop.getDesktop();
+            	desk.open(fileToOpen);
+            	model.getQueue().add(fileToOpen.getName());
+                JOptionPane.showMessageDialog(null, "File is opening successfully...");
+            } catch (IOException ex) {
+            }	 
+            
+        }else if(e.getActionCommand().equals("replace")) {

```

Comment:
Again, having a separate method for this chunk of code would it make it more maintainable, readable, and reusable.